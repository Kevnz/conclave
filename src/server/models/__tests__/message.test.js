const messageFixtures = require('../../../fixtures/messages-fixture.json')
const Message = require('../message')

describe('The Message Model', () => {
  it('should match the snapshopt', async () => {
    const message = await Message.where({ id: 1 }).fetch()
    const snapshotMessage = message.toJSON()
    delete snapshotMessage.created_at
    delete snapshotMessage.updated_at
    delete snapshotMessage.createdOn
    expect(snapshotMessage).toMatchSnapshot()
  })

  describe('Top Level', () => {
    it('should return the lop level (null parent_id) messages ', () => {
      return Message.collection()
        .query(qb => {
          qb.whereNull('parent_id')
        })
        .fetch()
        .then(messages => {
          const serializedMessages = messages.map(topic => topic.toJSON())
          const fixtureCount = messageFixtures.filter(
            topic => topic.parent_id === null
          ).length

          expect(serializedMessages.length === fixtureCount)
        })
    })
    it('should return the lop level (null parent_id) messages for a topic', () => {
      return Message.getByTopic(1).then(messages => {
        const serializedMessages = messages.map(message => message.toJSON())
        const fixtureCount = messageFixtures.filter(
          message => message.parent_id === null
        ).length
        expect(serializedMessages.length === fixtureCount)
      })
    })
    it('should return replies', () => {
      return Message.getReplies(11).then(messages => {
        const replies = messages.toJSON().length
        const fixtureRepliesLength = messageFixtures.filter(
          m => m.parent_id === 11
        ).length
        expect(replies).toBe(fixtureRepliesLength)
      })
    })
  })
})
