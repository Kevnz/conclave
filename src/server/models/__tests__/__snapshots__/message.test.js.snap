// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Message Model should match the snapshopt 1`] = `
Object {
  "body": "est et quis facere sint accusantium dolores sint debitis et neque quaerat omnis vel dolorem autem tempora quos voluptatem deleniti alias repudiandae exercitationem nobis consequatur reprehenderit perspiciatis voluptatem id minus est sapiente doloribus harum accusantium voluptas quidem repellat optio non",
  "created_at": 2019-03-14T18:25:03.789Z,
  "created_by": 1,
  "id": 1,
  "parent_id": null,
  "title": "Extended optimizing access",
  "topic_id": 3,
  "updated_at": 2019-03-14T18:25:03.789Z,
}
`;
